beluga_tutorial/README.md

```diff
@@ -0,0 +1,727 @@
+The estimation at any given time is represented using a probability distribution function, called **belief**, defined as:
+$$bel(x_t) = p(x_t|z_{1:t},u_{1:t})$$
+where $x_t$ is the state of the system at time $t$, and $z_{1:t}$ and $u_{1:t}$ are the sequence of sensor measurements and the sequence of control inputs up to time $t$ respectively. A belief reflects the robot's internal knowledge about the state of the environment. We therefore distinguish the **true state** from the robot's internal belief.
+
+It can be shown that the posterior belief can be recursively computed using the prior belief, the current sensor readings, and the current control inputs using the following update rule:
+$$bel(x_t)=η * p(z_t|x_t) * \int p(x_t|x_{t-1}, u_t)bel(x_{t-1}) \, dx_{t-1}$$
+where **η** is a normalization factor.
+
+A particle filter approximates this update rule by representing the belief as a set of discrete **samples**, each representing a possible state of the system. Collectively, these samples (or **particles**) represent the probability distribution over the state of the system at time $t$, condition on the prior state distribution, sensor readings, and control inputs.
+
+Compared to other variants of Bayesian state estimation algorithms, such as Kalman filters, particle filters have the advantage of being able
+to represent multimodal distributions and to easily incorporate complex dynamics and diverse sensor modalities.
+
+The purpose of this tutorial is to show a common implementation of the particle filter by performing the recursive update in three steps:
+* [Control Update](#214-control-update---motion-model)
+* [Measurement Update](#215-measurement-update---sensor-model)
+* [Resample](#216-resample)
+
+## Tasks
+
+## 1 Create the Workspace
+In this tutorial we are using the conventions of [ROS2](https://docs.ros.org/en/humble/index.html), so we need to create a workspace directory. Open a new terminal and create the directory:
+
+```bash
+mkdir -p beluga_ws/src
+```
+
+## 2 Create the Package
+In order to follow the [ROS2](https://docs.ros.org/en/humble/index.html) guidelines, we are going to create a package inside the `beluga_ws/src` path. Therefore, let's start by creating the necessary directories. In a terminal, run the following command:
+
+```bash
+mkdir -p beluga_ws/src/beluga_tutorial/src
+```
+
+`beluga_tutorial` should have a `package.xml` and a `CmakeLists.txt` in order to be able to use `colcon` for the compilation process. We will create these files later ([for more details](#4-compile-and-run-code)).
+
+## 3 Write the Simulation code
+To build the simulation step by step, we'll first explain the main functionality of the code, which is assembling the particle filter, and all the necessary resources around this. In a second step, we'll explain the necessary functionalities for a post-processing simulation's information.
+
+To start, create a `beluga_tutorial/src/main.cpp` file and put the following code inside:
+
+```cpp
```

Comment:
meta: I think we can skip the YAML serialization and de-serialization code here. We can point them to the code checked in the repository for a complete example with full visualization support.

It would be good to focus solely on the particle filter implementation which shows code that compiles but doesn't necessarily generate any output.

That said, it would be great to support the different concepts and implementation explanation with images generated from our visualization tool.

Reply:
Okay, I agree with you, I'll change the code here to use the default values of the Parameters structer and I'll erase all the YAML related code.